From d41a453bbf195818e3fa66ed0a123f0c06aa754b Mon Sep 17 00:00:00 2001
From: Arun Sharma <asharma@fb.com>
Date: Sun, 19 May 2013 00:04:15 -0700
Subject: [PATCH] Fix a build error

<elf.h> needs to be included before the check for ELF_CLASS.
---
 src/elfxx.h |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/elfxx.h b/src/elfxx.h
index 558a235..f3917ee 100644
--- a/src/elfxx.h
+++ b/src/elfxx.h
@@ -30,6 +30,8 @@ WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.  */
 #include <sys/mman.h>
 #include <sys/stat.h>
 
+#include "libunwind_i.h"
+
 #if ELF_CLASS == ELFCLASS32
 # define ELF_W(x)	ELF32_##x
 # define Elf_W(x)	Elf32_##x
@@ -40,8 +42,6 @@ WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.  */
 # define elf_w(x)	_Uelf64_##x
 #endif
 
-#include "libunwind_i.h"
-
 extern int elf_w (get_proc_name) (unw_addr_space_t as,
 				  pid_t pid, unw_word_t ip,
 				  char *buf, size_t len,
-- 
1.7.2.5

